orchestration/templates/ingest-encode-data.yaml

```diff
@@ -35,3 +35,39 @@ spec:
                   value: '{{ "{{inputs.parameters.gcs-prefix}}" }}/raw'
                 - name: output-prefix
                   value: '{{ "{{inputs.parameters.gcs-prefix}}" }}/processed'
+          {{- $datasetName := printf "%s_%s" .Values.bigquery.stagingData.datasetPrefix (include "argo.timestamp" .) }}
+          # 3) Diff generated data against existing data
+          - name: process-tables
+            dependencies: [run-transformation]
+            templateRef:
+              name: process-table
+              template: main
+            arguments:
+              parameters:
+                - name: table-name
+                  value: '{{ "{{item}}" }}'
+                - name: gcs-bucket
+                  value: '{{ .Values.gcs.bucketName }}'
+                - name: gcs-prefix
+                  value: '{{ "{{inputs.parameters.gcs-prefix}}" }}'
+                - name: staging-bq-project
+                  value: '{{ .Values.bigquery.stagingData.project }}'
+                - name: staging-bq-dataset
+                  value: {{ $datasetName }}
+                {{- with .Values.bigquery.jadeData }}
+                - name: jade-bq-project
+                  value: '{{ .project }}'
+                - name: jade-bq-dataset
+                  value: '{{ .dataset }}'
+                {{- end }}
+            withItems:
```

Comment:
This block is specifying the tables to loop over for this step of the DAG, right?
So this would run the process-table template once for every item as an input param for table-name?
Just wanted to confirm my understanding!

Reply:
Yeah, that's my basic understanding too. Each of the table names is passed in as a parameter using the `{{item}}` variable and I think there is an async task launched for each of them.